### ISSUE TITLE
User-space stack backtrace

### ISSUE THREAD

**Issue by hardc0der** [iovisor/bcc#2386]
This was kind of discussed in [issue #1070](https://github.com/iovisor/bcc/issues/1070) but I'm having trouble getting user-space backtraces from 32-bit ARM binaries. If I take a simple test application I get proper user space callchains for the 64-bit binary but not the 32-bit binary. For the 32-bit binary I just get one instruction pointer/program counter so it seems like the in-kernel unwinder has trouble. The unwind information is stored a bit differently in 32-bit ARM binaries so I wonder if that's the problem. Are there any workarounds? @4ast mentioned possibly reading some of the stack memory and unwinding in userspace like perf allows you to do. Are there any examples of that in bcc?

**Comment by yonghong-song** [comment]
Did you compile with `-fno-omit-frame-pointer` on your 32-bit ARM platform?
Without frame pointer, current kernel won't be able to reliably get the user space stack trace.

**Comment by hardc0der** [comment]
No, I want it to work with just `-funwind-tables` for the 32-bit code. It seems like I'll need to do what `perf` does for 32-bit ARM and send some of the stack to userspace for unwinding using `libunwind`. It seems like that's what @4ast suggested in the previous issue so I was looking for any examples of reading the stack.

**Comment by yonghong-song** [comment]
Okay, you then need to read the memory and send them to user space for further analysis.
